bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/dialogs/ViewsPreferencePage.java

```diff
@@ -127,6 +132,8 @@ protected Control createContents(Composite parent) {
 		Composite comp = new Composite(parent, SWT.NONE);
 
 		themingEnabled = createCheckButton(comp, WorkbenchMessages.ThemingEnabled, engine != null);
+		initialStateUpdateOnRuntime = PrefUtil.getAPIPreferenceStore()
+				.getBoolean(IWorkbenchPreferenceConstants.RESCALING_AT_RUNTIME);
```

Comment:
As mentioned in https://github.com/eclipse-platform/eclipse.platform.ui/pull/2461#discussion_r1824038278, I do not yet understand why we need to store this information in a field instead of retrieving it when required, like done for the themeing.

Reply:
So that it could be used across the class. I needed this field info while creating the HiDpi setting group and also when Ok is performed. For theming they checked the initial state by simply checking if theme != null. I hope it make sense

Reply:
Now, I am retrieving the value only in the methods where its needed, separately.